src/Razor/src/Microsoft.AspNetCore.Razor.VSCode.BlazorWasmDebuggingExtension/src/extension.ts

```diff
@@ -25,46 +25,51 @@ export function activate(context: vscode.ExtensionContext) {
-            for await (const output of spawnedProxy.stdout) {
-                // If we haven't found the URL in the first ten chunks processed
-                // then bail out.
-                if (chunksProcessed++ > 10) {
-                    return;
+            try {
+                let chunksProcessed = 0;
+                for await (const output of spawnedProxy.stdout) {
+                    // If we haven't found the URL in the first ten chunks processed
+                    // then bail out.
+                    if (chunksProcessed++ > 10) {
+                        return;
+                    }
+                    outputChannel.appendLine(output);
+                    // The debug proxy server outputs the port it is listening on in the
+                    // standard output of the launched application. We need to pass this URL
+                    // back to the debugger so we extract the URL from stdout using a regex.
+                    // The debug proxy will not exit until killed via the `killDebugProxy`
+                    // method so parsing stdout is necessary to extract the URL.
+                    const matchExpr = 'Now listening on: (?<url>.*)';
+                    const found = `${output}`.match(matchExpr);
+                    const url = found?.groups?.url;
+                    if (url) {
+                        outputChannel.appendLine(`Debugging proxy is running at: ${url}`);
+                        pidsByUrl.set(url, spawnedProxy.pid);
+                        return {
+                            url,
+                            inspectUri: `${url}{browserInspectUriPath}`,
+                            debuggingPort,
+                        };
+                    }
                 }
-                outputChannel.appendLine(output);
-                // The debug proxy server outputs the port it is listening on in the
-                // standard output of the launched application. We need to pass this URL
-                // back to the debugger so we extract the URL from stdout using a regex.
-                // The debug proxy will not exit until killed via the `killDebugProxy`
-                // method so parsing stdout is necessary to extract the URL.
-                const matchExpr = 'Now listening on: (?<url>.*)';
-                const found = `${output}`.match(matchExpr);
-                const url = found?.groups?.url;
-                if (url) {
-                    outputChannel.appendLine(`Debugging proxy is running at: ${url}`);
-                    pidsByUrl.set(url, spawnedProxy.pid);
-                    return {
-                        url,
-                        inspectUri: `${url}{browserInspectUriPath}`,
-                        debuggingPort,
-                    };
+
+                for await (const error of spawnedProxy.stderr) {
+                    outputChannel.appendLine(`ERROR: ${error}`);
+                    process.kill(spawnedProxy.pid);
```

Comment:
> If spawnProxy logs two errors right after each other won't this only log the first one

Ah good catch. This is in line with [current behavior](https://github.com/dotnet/razor-tooling/pull/5885/files/b0d4f5a38e5f4ef451b381e4fe815bfac9cf4aa5#diff-aa3450f1f9095d42ff8276c927d38d957d74b81ba46334c17dcbbd71a96b953dR5), but I can see how this could be problematic. Updated.

> then try to read from stderr again

We return on the next line so I don't believe it'd try to continue reading here.